lib/awful/layout/init.lua

```diff
@@ -162,6 +162,11 @@ function layout.arrange(screen)
     delayed_arrange[screen] = true
 
     timer.delayed_call(function()
+        if not screen.valid then
```

Comment:
Hm, if you have a test for `tests/` for that, I'd like to check. I just added the following to the "catch all" `awful.rules`-rule in the default config and ran the test suite. It printed a bunch of true/false, but no errors.
```
callback = function(c) gears.timer.start_new(0.001, function() print(c.valid) return true end) end},
```